### DISCUSSION TITLE
Nvim fails to read system clipboard in wayland

### DISCUSSION THREAD

**Discussion by ChenZhongPu** [LazyVim/LazyVim#4358] [Q&A] [answered]
I know this question has nothing to do with lazyvim, and I am here just for help.

I found that in Wayland, nvim fails to read system clipboard, so I cannot paste using `p`, and I have to use `shift+ctrl+v`, how to solve it?  Here is the workflow:

- Step 1: Copy some texts in some applications (`ctrl-c`)
- Step 2: Try to paste in nvim.

I also checked the content in `+` register, and it is NOT the content in system clipboard.

On the other hand, yanking texts in nvim can be found in system clipboard. Also, I notice there is a message,

> clipboard: error: Error: target STRING not available.

The following is the output of `checkhealth`:

> provider.clipboard: require("provider.clipboard.health").check()
Clipboard (optional) ~
>- OK Clipboard tool found: xclip

**Comment by ChenZhongPu** [comment, marked as answer]
It is solved after installing `wl-clipboard`.

**Reply by Torsteinws** [reply]
Thank you! This solved the problem for me too.

I feel like this should be documented somewhere. The solution might be obvious for experienced users, but it is for sure not obvious for someone like me who is new to linux and nvim.
